core/trino-main/src/main/java/io/trino/execution/buffer/PagesSerdeUtil.java

```diff
@@ -124,29 +130,53 @@ protected Page computeNext()
         }
     }
 
-    public static Iterator<Slice> readSerializedPages(SliceInput sliceInput)
+    public static Iterator<Slice> readSerializedPages(LittleEndianDataInputStream inputStream)
     {
-        return new SerializedPageReader(sliceInput);
+        return new SerializedPageReader(inputStream);
     }
 
     private static class SerializedPageReader
             extends AbstractIterator<Slice>
     {
-        private final SliceInput input;
+        private final LittleEndianDataInputStream input;
```

Comment:
nit: `LittleEndianDataInputStream` is not efficient at decoding `int` / `long`. A more efficient way would be to allocate a single `Slice` of a size of the header. Then read the entire header from an InputStream at once into a `Slice` and extract the page size as `int` with `Slice#getInt` that does that in a single machine instruction. Then the content of that `Slice` could be efficiently copied into a `Slice` returned from the `computeNext` method.

Reply:
Make sense, addressed comment.